babeldoc: init at 0.5.22

ryota2357 opened the pull request:
BabelDOC is a PDF scientific paper translation and bilingual comparison library.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin (build with `--system x86_64-darwin`)
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ryota2357 committed:
babeldoc: init at 0.5.22

dotlambda commented on pkgs/development/python-modules/babeldoc/default.nix:
```diff
@@ -0,0 +1,121 @@
+    homepage = "https://github.com/funstory-ai/BabelDOC";
+    platforms = [
+      "x86_64-linux"
+      "x86_64-darwin"
```
Why so restrictive? I don't see any non-Python code in the repo, so any restrictive `platforms` should be specified by dependencies.

dotlambda reviewed (commented):
Please add pytestCheckHook and versionCheckHook.

ryota2357 commented on pkgs/development/python-modules/babeldoc/default.nix:
```diff
@@ -0,0 +1,121 @@
+  scikit-learn,
+}:
+
+buildPythonPackage rec {
```
Moved to by-name and fixed to use `buildPythonApplication`.

ryota2357 commented:
@dotlambda Thank you for the review! I've fixed and updated.

> Please add pytestCheckHook and versionCheckHook.

If added `pytestCheckHook` cause build error.

<details>
<summary>build log (if added pythonCheckHook)</summary>

```
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pypa-build-hook
Using pypaBuildPhase
Sourcing python-runtime-deps-check-hook
Using pythonRuntimeDepsCheckHook
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing pytest-check-hook
Using pytestCheckPhase
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/8drzxbjdj233cc93fwhgadyqiwi2risx-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file "source/uv.lock"
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing pypaBuildPhase
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
* Getting build dependencies for wheel...
* Building wheel...
Successfully built babeldoc-0.5.21-py3-none-any.whl
Finished creating a wheel...
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
@nix { "action": "setPhase", "phase": "pythonRuntimeDepsCheckHook" }
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for babeldoc-0.5.21-py3-none-any.whl
Finished executing pythonRuntimeDepsCheck
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Executing pypaInstallPhase
Successfully installed babeldoc-0.5.21-py3-none-any.whl
Finished executing pypaInstallPhase
Running phase: pythonOutputDistPhase
@nix { "action": "setPhase", "phase": "pythonOutputDistPhase" }
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
Running phase: fixupPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
checking for references to /nix/var/nix/builds/nix-11634-1144407885/ in /nix/store/4113nxg1ayl3j89brg6fwfp8p4x5hdib-babeldoc-0.5.21...
patching script interpreter paths in /nix/store/4113nxg1ayl3j89brg6fwfp8p4x5hdib-babeldoc-0.5.21
/nix/store/4113nxg1ayl3j89brg6fwfp8p4x5hdib-babeldoc-0.5.21/lib/python3.13/site-packages/babeldoc/pdfminer/psparser.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/84s01mzl74vh69aphpl3rf07pp49pkz0-python3-3.13.9/bin/python3"
stripping (with command strip and flags -S) in  /nix/store/4113nxg1ayl3j89brg6fwfp8p4x5hdib-babeldoc-0.5.21/lib /nix/store/4113nxg1ayl3j89brg6fwfp8p4x5hdib-babeldoc-0.5.21/bin
checking for references to /nix/var/nix/builds/nix-11634-1144407885/ in /nix/store/kns8d4f4n2sinaq0z8ih3vc63nckypza-babeldoc-0.5.21-dist...
patching script interpreter paths in /nix/store/kns8d4f4n2sinaq0z8ih3vc63nckypza-babeldoc-0.5.21-dist
Rewriting #!/nix/store/84s01mzl74vh69aphpl3rf07pp49pkz0-python3-3.13.9/bin/python3.13 to #!/nix/store/84s01mzl74vh69aphpl3rf07pp49pkz0-python3-3.13.9
wrapping `/nix/store/4113nxg1ayl3j89brg6fwfp8p4x5hdib-babeldoc-0.5.21/bin/babeldoc'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
@nix { "action": "setPhase", "phase": "installCheckPhase" }
no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
@nix { "action": "setPhase", "phase": "pythonCatchConflictsPhase" }
Running phase: pythonRemoveBinBytecodePhase
@nix { "action": "setPhase", "phase": "pythonRemoveBinBytecodePhase" }
Running phase: pythonImportsCheckPhase
@nix { "action": "setPhase", "phase": "pythonImportsCheckPhase" }
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: babeldoc
Running phase: pytestCheckPhase
@nix { "action": "setPhase", "phase": "pytestCheckPhase" }
Executing pytestCheckPhase
pytest flags: -m pytest
============================= test session starts ==============================
platform darwin -- Python 3.13.9, pytest-8.4.2, pluggy-1.6.0
rootdir: /nix/var/nix/builds/nix-11634-1144407885/source
configfile: pyproject.toml
testpaths: tests
plugins: anyio-4.11.0
collecting ... 
collected 0 items / 1 error                                                    

==================================== ERRORS ====================================
___________ ERROR collecting tests/test_translation_cache_cleanup.py ___________
/nix/store/84s01mzl74vh69aphpl3rf07pp49pkz0-python3-3.13.9/lib/python3.13/pathlib/_local.py:722: in mkdir
    os.mkdir(self, mode)
E   FileNotFoundError: [Errno 2] No such file or directory: '/homeless-shelter/.cache/babeldoc/tiktoken'

During handling of the above exception, another exception occurred:
/nix/store/84s01mzl74vh69aphpl3rf07pp49pkz0-python3-3.13.9/lib/python3.13/pathlib/_local.py:722: in mkdir
    os.mkdir(self, mode)
E   FileNotFoundError: [Errno 2] No such file or directory: '/homeless-shelter/.cache/babeldoc'

During handling of the above exception, another exception occurred:
/nix/store/84s01mzl74vh69aphpl3rf07pp49pkz0-python3-3.13.9/lib/python3.13/pathlib/_local.py:722: in mkdir
    os.mkdir(self, mode)
E   FileNotFoundError: [Errno 2] No such file or directory: '/homeless-shelter/.cache'

During handling of the above exception, another exception occurred:
tests/test_translation_cache_cleanup.py:3: in <module>
    from babeldoc.translator.cache import TranslationCache
babeldoc/translator/cache.py:16: in <module>
    from babeldoc.const import CACHE_FOLDER
babeldoc/const.py:43: in <module>
    TIKTOKEN_CACHE_FOLDER.mkdir(parents=True, exist_ok=True)
/nix/store/84s01mzl74vh69aphpl3rf07pp49pkz0-python3-3.13.9/lib/python3.13/pathlib/_local.py:726: in mkdir
    self.parent.mkdir(parents=True, exist_ok=True)
/nix/store/84s01mzl74vh69aphpl3rf07pp49pkz0-python3-3.13.9/lib/python3.13/pathlib/_local.py:726: in mkdir
    self.parent.mkdir(parents=True, exist_ok=True)
/nix/store/84s01mzl74vh69aphpl3rf07pp49pkz0-python3-3.13.9/lib/python3.13/pathlib/_local.py:726: in mkdir
    self.parent.mkdir(parents=True, exist_ok=True)
/nix/store/84s01mzl74vh69aphpl3rf07pp49pkz0-python3-3.13.9/lib/python3.13/pathlib/_local.py:722: in mkdir
    os.mkdir(self, mode)
E   OSError: [Errno 30] Read-only file system: '/homeless-shelter'
=========================== short test summary info ============================
ERROR tests/test_translation_cache_cleanup.py - OSError: [Errno 30] Read-only file system: '/homeless-shelter'
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.29s ===============================
```

</details>


<details>
<summary>build log (if added versionCheckHook)</summary>

```
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pypa-build-hook
Using pypaBuildPhase
Sourcing python-runtime-deps-check-hook
Using pythonRuntimeDepsCheckHook
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Using versionCheckHook
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/8drzxbjdj233cc93fwhgadyqiwi2risx-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file "source/uv.lock"
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing pypaBuildPhase
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
[1m* Getting build dependencies for wheel...[0m
[1m* Building wheel...[0m
[1m[92mSuccessfully built [4mbabeldoc-0.5.21-py3-none-any.whl[0m[1m[92m[0m
Finished creating a wheel...
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
@nix { "action": "setPhase", "phase": "pythonRuntimeDepsCheckHook" }
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for babeldoc-0.5.21-py3-none-any.whl
Finished executing pythonRuntimeDepsCheck
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Executing pypaInstallPhase
Successfully installed babeldoc-0.5.21-py3-none-any.whl
Finished executing pypaInstallPhase
Running phase: pythonOutputDistPhase
@nix { "action": "setPhase", "phase": "pythonOutputDistPhase" }
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
Running phase: fixupPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
checking for references to /nix/var/nix/builds/nix-81578-1604113615/ in /nix/store/5nvz27njvc4m4h23j8zr3r4p26j7w5z7-babeldoc-0.5.21...
patching script interpreter paths in /nix/store/5nvz27njvc4m4h23j8zr3r4p26j7w5z7-babeldoc-0.5.21
/nix/store/5nvz27njvc4m4h23j8zr3r4p26j7w5z7-babeldoc-0.5.21/lib/python3.13/site-packages/babeldoc/pdfminer/psparser.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/84s01mzl74vh69aphpl3rf07pp49pkz0-python3-3.13.9/bin/python3"
stripping (with command strip and flags -S) in  /nix/store/5nvz27njvc4m4h23j8zr3r4p26j7w5z7-babeldoc-0.5.21/lib /nix/store/5nvz27njvc4m4h23j8zr3r4p26j7w5z7-babeldoc-0.5.21/bin
checking for references to /nix/var/nix/builds/nix-81578-1604113615/ in /nix/store/45mdmy5igscpfy6gpqhn5rz3jvvsr7kg-babeldoc-0.5.21-dist...
patching script interpreter paths in /nix/store/45mdmy5igscpfy6gpqhn5rz3jvvsr7kg-babeldoc-0.5.21-dist
Rewriting #!/nix/store/84s01mzl74vh69aphpl3rf07pp49pkz0-python3-3.13.9/bin/python3.13 to #!/nix/store/84s01mzl74vh69aphpl3rf07pp49pkz0-python3-3.13.9
wrapping `/nix/store/5nvz27njvc4m4h23j8zr3r4p26j7w5z7-babeldoc-0.5.21/bin/babeldoc'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
@nix { "action": "setPhase", "phase": "installCheckPhase" }
Executing versionCheckPhase
Did not find version 0.5.21 in the output of the command /nix/store/5nvz27njvc4m4h23j8zr3r4p26j7w5z7-babeldoc-0.5.21/bin/babeldoc --version
Traceback (most recent call last):
  File "{{storeDir}}/lib/python3.13/pathlib/_local.py", line 722, in mkdir
    os.mkdir(self, mode)
    ~~~~~~~~^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/var/empty/.cache/babeldoc/tiktoken'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "{{storeDir}}/lib/python3.13/pathlib/_local.py", line 722, in mkdir
    os.mkdir(self, mode)
    ~~~~~~~~^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/var/empty/.cache/babeldoc'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "{{storeDir}}/bin/.babeldoc-wrapped", line 6, in <module>
    from babeldoc.main import cli
  File "{{storeDir}}/lib/python3.13/site-packages/babeldoc/main.py", line 19, in <module>
    import babeldoc.assets.assets
  File "{{storeDir}}/lib/python3.13/site-packages/babeldoc/assets/assets.py", line 25, in <module>
    from babeldoc.const import get_cache_file_path
  File "{{storeDir}}/lib/python3.13/site-packages/babeldoc/const.py", line 43, in <module>
    TIKTOKEN_CACHE_FOLDER.mkdir(parents=True, exist_ok=True)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "{{storeDir}}/lib/python3.13/pathlib/_local.py", line 726, in mkdir
    self.parent.mkdir(parents=True, exist_ok=True)
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "{{storeDir}}/lib/python3.13/pathlib/_local.py", line 726, in mkdir
    self.parent.mkdir(parents=True, exist_ok=True)
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "{{storeDir}}/lib/python3.13/pathlib/_local.py", line 722, in mkdir
    os.mkdir(self, mode)
    ~~~~~~~~^^^^^^^^^^^^
PermissionError: [Errno 1] Operation not permitted: '/var/empty/.cache'
```

</details>

ryota2357 commented on pkgs/development/python-modules/babeldoc/default.nix:
```diff
@@ -0,0 +1,121 @@
+  patches = [
+    (fetchpatch2 {
+      name = "rename-python-levenshtein-to-levenshtein";
+      url = "https://github.com/funstory-ai/BabelDOC/pull/542.patch";
```
PR was merged, so I use `pathces`.

ryota2357 commented on pkgs/by-name/ba/babeldoc/package.nix:
```diff
@@ -0,0 +1,121 @@
+
+  pythonImportsCheck = [ "babeldoc" ];
+
+  meta = {
```
Added the link, thank you!

ryota2357 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 466475`
Commit: [`2a4a2297ff1e891962becef873e3a283a0b67761`](https://github.com/NixOS/nixpkgs/commit/2a4a2297ff1e891962becef873e3a283a0b67761) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/2a4a2297ff1e891962becef873e3a283a0b67761..pull/466475/head))
Merge: [`75040cead9f0ebbd2c9721c37081ec774f9461cb`](https://github.com/NixOS/nixpkgs/commit/75040cead9f0ebbd2c9721c37081ec774f9461cb)

Logs: https://github.com/ryota2357/nixpkgs-review-gha/actions/runs/19848689883


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>babeldoc</li>
    <li>babeldoc.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
:white_check_mark: *No rebuilds*

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>babeldoc</li>
    <li>babeldoc.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
:white_check_mark: *No rebuilds*

Aleksanaa commented on pkgs/by-name/ba/babeldoc/package.nix:
```diff
@@ -0,0 +1,96 @@
+  versionCheckKeepEnvironment = "HOME";
+
+  meta = {
+    description = "PDF scientific paper translation and bilingual comparison library.";
```
```suggestion
    description = "PDF scientific paper translation and bilingual comparison library";
```

dotlambda commented on pkgs/by-name/ba/babeldoc/package.nix:
```diff
@@ -0,0 +1,96 @@
+  patches = [
+    (fetchpatch2 {
+      name = "rename-python-levenshtein-to-levenshtein";
+      url = "https://github.com/funstory-ai/BabelDOC/pull/542.patch";
```
use `fetchpatch` or `?full_index=1`

ryota2357 replied:
Added `?full_index=1`.

dotlambda commented on pkgs/by-name/ba/babeldoc/package.nix:
```diff
@@ -0,0 +1,96 @@
+    python3Packages.pytestCheckHook
+    versionCheckHook
+  ];
+  versionCheckProgramArg = "--version";
```
```suggestion
```

dotlambda commented on pkgs/by-name/ba/babeldoc/package.nix:
```diff
@@ -0,0 +1,96 @@
+  meta = {
+    description = "PDF scientific paper translation and bilingual comparison library";
+    homepage = "https://github.com/funstory-ai/BabelDOC";
+    changelog = "https://github.com/funstory-ai/BabelDOC/releases/tag/v${version}";
```
```suggestion
    changelog = "https://github.com/funstory-ai/BabelDOC/releases/tag/${src.tag}";
```

ryota2357 replied:
Fixed

dotlambda commented on pkgs/by-name/ba/babeldoc/package.nix:
```diff
@@ -0,0 +1,90 @@
+  meta = {
+    description = "PDF scientific paper translation and bilingual comparison library";
+    homepage = "https://github.com/funstory-ai/BabelDOC";
+    changelog = "https://github.com/funstory-ai/BabelDOC/releases/tag/${src.version}";
```
```suggestion
    changelog = "https://github.com/funstory-ai/BabelDOC/releases/tag/${src.tag}";
```

ryota2357 replied:
Sorry, I've fixed.

dotlambda commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 466475`
Commit: `946047480bf4cd10d21da030b5f1b9613318aabe`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>babeldoc</li>
    <li>babeldoc.dist</li>
  </ul>
</details>

dotlambda commented:
One last thing: the commit message shouldn't say `python3Packages.`
